source/extensions/common/proxy_protocol/proxy_protocol_header.cc

```diff
@@ -111,12 +111,26 @@ void generateV2Header(const Network::Address::Ip& source_address,
 }
 
 bool generateV2Header(const Network::ProxyProtocolData& proxy_proto_data, Buffer::Instance& out,
-                      bool pass_all_tlvs, const absl::flat_hash_set<uint8_t>& pass_through_tlvs) {
-  uint64_t extension_length = 0;
-  for (auto&& tlv : proxy_proto_data.tlv_vector_) {
+                      bool pass_all_tlvs, const absl::flat_hash_set<uint8_t>& pass_through_tlvs,
+                      const std::unordered_map<uint8_t, std::vector<unsigned char>>& custom_tlvs) {
```

Comment:
I decided to make the custom_tlvs parameter a map instead of vector -- largely to prevent duplicate TLV keys between the passthrough & custom TLV keyset -- for quick O(1) key lookups.

Reply:
these small scale vector perf is probably better

Reply:
Definitely agree perf would be better with a small vector here. I ran into several issues with previous implementations that tried to define a vector. Mainly because I wanted to re-use the ProxyProtocolTLV type:

```cpp
struct ProxyProtocolTLV {
  const uint8_t type;
  const std::vector<unsigned char> value;
};

using ProxyProtocolTLVVector = std::vector<ProxyProtocolTLV>;
```

IIRC, builds were failing and complaining that copy assignment was disabled for that type. I tried to evade that issue several times by using std::move, emplace, passing the parameter as a const ref, etc. but I wasn't able to get builds back online. Any idea on what I'd need to do there?

Reply:
i would have worded my comment better; as the number of items here will likely be less than 10, i don't think it matters if we use map or vector.